src/classes/TDTM_Filter.cls

```diff
@@ -414,12 +414,20 @@ public class TDTM_Filter {
             return Date.parse(val);
         } else if(fieldType == DisplayType.Reference) {
             return ID.valueOf(val);
+        } else if(fieldType == DisplayType.CURRENCY) {
+            return Decimal.valueOf(val);
+        } else if(fieldType == DisplayType.INTEGER) {
+            return Integer.valueOf(val);
+        } else if(fieldType == DisplayType.DOUBLE) {
+            return Double.valueOf(val);
+        } else if(fieldType == DisplayType.PERCENT) {
+            return Decimal.valueOf(val);
```

Comment:
Ideally we need the unit test to test the Numeric field.  Can we add some test cases for one/more new supported type? Maybe we can use some Account Standard field for filtering.